File: Methods-C++/chebyshevDistance.cpp
#include <iostream>

double chebyshevDistance(double p1[], double p2[], int n1, int n2){
    if (n1 !=  n2)
    {
        cout << "Error!" << endl;
        return 0;
    }

    double maxDiff = abs(p1[0] - p2[0]);
    for (int i = 1; i < n1; i++)
    {
        double diff = p1[i] - p2[i];
        if (maxDiff < diff)
        {
            maxDiff = diff;
        }
    }
    return maxDiff;        
}

File: Methods/Methods-Python/chebyshevDistance.py
def chebyshevDistance(p1, p2):
    if len(p1) != len(p2):
        print('Error!')
        return -1

    maxDiff = abs(p1[0] - p2[0])
    for i in range(0, len(p1)):
        diff = abs(p1[i] - p2[i])

        if maxDiff < diff:
            maxDiff = diff

    return maxDiff
